### PR TITLE
:green_heart: Fix compilation of Python wheels for Windows

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

Github has released an update to MSVC v144 for the ``windows-latest`` runner. This could cause this problem as discussed [here](https://github.com/pypa/cibuildwheel/issues/1875). So we have to use ``windows-2019`` until this is fixed.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [ ] I have added appropriate tests and documentation.
- [ ] I have added a changelog entry.
- [ ] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

**Comment by Drewniok** [comment]
@marcelwa After spending several hours, I can now say that the problem is caused by multithreading in _QuickSim_. I have tried to solve the problem by using ``std::future`` without success. 
The problem is solved as soon as multithreading is not used. I think this is quite an important issue and topic since multithreading is important for our tools (_QuickSim_, _OpDomain_, etc.).

**Comment by marcelwa** [comment]
I agree. What exactly is it about the multi-threading that causes this problem do you think?

**Comment by marcelwa** [comment]
Have you tried activating the respective sanitizers?

File: .github/workflows/python-bindings.yml [hunk 1]
```diff
@@ -69,7 +75,7 @@ jobs:
       - name: Setup ccache
         uses: hendrikmuhs/ccache-action@v1.2
         with:
-          key: '${{matrix.os}}-pyfiction'
+          key: '${{matrix.os}}-${{matrix.toolset}}-pyfiction'
```
Comment (marcelwa) [inline .github/workflows/python-bindings.yml:78]: I don't think you need the toolset key because there is only one build per operating system

File: .github/workflows/pyfiction-pypi-deployment.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
           - { os: ubuntu-latest, arch: x64 }
           - { os: macos-latest, arch: x64 }
           - { os: macos-latest, arch: arm64 }
-          - { os: windows-latest, arch: x64 }
+          - { os: windows-2019, arch: x64}
```
Comment (marcelwa) [inline .github/workflows/pyfiction-pypi-deployment.yml:56]:
```suggestion
          - { os: windows-2019, arch: x64 }
```

File: .github/workflows/pyfiction-pypi-deployment.yml [hunk 3]
```diff
@@ -53,7 +53,7 @@ jobs:
           - { os: ubuntu-latest, arch: x64 }
           - { os: macos-latest, arch: x64 }
           - { os: macos-latest, arch: arm64 }
```
Comment (marcelwa) [inline .github/workflows/pyfiction-pypi-deployment.yml:55]: Is there a reason for not specifying the exact versions here, too?

File: bindings/pyfiction/include/pyfiction/algorithms/simulation/sidb/quicksim.hpp [hunk 4]
```diff
@@ -44,9 +44,7 @@ inline void quicksim(pybind11::module& m)
                        DOC(fiction_quicksim_params_iteration_steps))
         .def_readwrite("alpha", &fiction::quicksim_params::alpha, DOC(fiction_quicksim_params_alpha))
         .def_readwrite("number_threads", &fiction::quicksim_params::number_threads,
-                       DOC(fiction_quicksim_params_number_threads))
-
-        ;
+                       DOC(fiction_quicksim_params_number_threads));
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/algorithms/simulation/sidb/quicksim.hpp:47]: Please revert this change
```suggestion
                       DOC(fiction_quicksim_params_number_threads))
                       
                       ;
```

File: test/benchmark/simulation.cpp [hunk 5]
```diff
@@ -61,7 +62,7 @@ TEST_CASE("Benchmark simulators", "[benchmark]")
 
     BENCHMARK("QuickExact")
     {
-        const quickexact_params<lattice> sim_params{sidb_simulation_parameters{2, -0.32}};
+        const quickexact_params<cell<lattice>> sim_params{sidb_simulation_parameters{2, -0.32}};
```
Comment (marcelwa) [inline test/benchmark/simulation.cpp:65]: If this change was necessary, how did it compile before? This looks like something where only one of the solutions would compile.

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: test/benchmark/simulation.cpp (hunk 5 above)
Comment (Drewniok) [reply test/benchmark/simulation.cpp:65]: Yes, the reason is that the benchmark code is not compiled for the CI. Should we change that?

File: .github/workflows/pyfiction-pypi-deployment.yml (hunk 3 above)
Comment (Drewniok) [reply .github/workflows/pyfiction-pypi-deployment.yml:55]: No, I will update this part.

File: test/benchmark/simulation.cpp (hunk 5 above)
Comment (marcelwa) [reply test/benchmark/simulation.cpp:65]: Good point. If we do we need to exclude its execution in the test run. Otherwise, this would just consume too much time. I'll take a note of this. Later problem 😄

Comment (Drewniok) [reply test/benchmark/simulation.cpp:65]: Yes, that's the point. Then, we just leave it for now. Agreed! 😁
